### PR TITLE
CRM-16294: Support map loading under https

### PR DESCRIPTION
**Pull request by yourcelf**
Replace http://openlayers/api/OpenLayers.js with CDNjs, to allow loading 
maps when civicrm is used with https.  Switch the default tile layer from 
openstreetmaps.org to MapQuest open tiles, which supports https.

https://issues.civicrm.org/jira/browse/CRM-16294

### CODE REVIEW

**Comment by colemanw** [comment]
jenkins ok to test

File: templates/CRM/Contact/Form/Task/Map/OpenStreetMaps.tpl [hunk 1]
```diff
@@ -32,7 +32,7 @@
 {/if}
 {assign var=defaultZoom value=12}
 {literal}
-<script src="http://openlayers.org/api/OpenLayers.js" type="text/javascript"></script>
+<script src="https://cdnjs.cloudflare.com/ajax/libs/openlayers/2.13.1/OpenLayers.js" type="text/javascript"></script>
```
Comment (colemanw) [inline templates/CRM/Contact/Form/Task/Map/OpenStreetMaps.tpl]: is there any reason to specify protocol here? Seems like it might be better to just start with `//`

Comment (yourcelf) [inline templates/CRM/Contact/Form/Task/Map/OpenStreetMaps.tpl]: Protocol-relative URLs can be [considered an anti-pattern](http://www.paulirish.com/2010/the-protocol-relative-url/) , as there are no circumstances under which retrieving from HTTPS is a problem.  Privacy organizations (EFF, ACLU, etc) universally recommend using HTTPS wherever possible, and major browser vendors (Google, Mozilla, etc) are making rumblings about deprecating unencrypted HTTP.  Using https here ensures that the OpenLayers script is not tampered with, and protects the privacy of people requesting the script (e.g. revealing which script on cdnjs.cloudflare.com is being retrieved).

In short: there's nothing in particular to gain from a protocol-relative URL, a bit of clarity to gain from specifying the protocol explicitly, and a teensy-tiny nod toward privacy to use https even when the parent site is hosted under http.

**Comment by colemanw** [comment]
That makes sense. Thanks for the PR.
